torch/csrc/jit/backends/backend_init.cpp

```diff
@@ -0,0 +1,181 @@
+              cloned_module._ivalue(),
+              toIValue(method_compile_spec, any_dict_ty).toGenericDict());
+          preprocess_fn->run(stack);
+          auto preprocessed_module = pop(stack);
+
+          // Create backend instance.
+          push(stack, IValue(preprocessed_module));
+          fn->run(stack);
+          auto backend = pop(stack).toCapsule();
+          TORCH_INTERNAL_ASSERT(backend);
+
+          // Generate LoweredModule.
+          Module loweredModule("torch.jit." + backend_name + "LoweredModule");
+
+          // Generate attributes.
+
+          // This is the original cloned and preprocessed module.
+          loweredModule.register_attribute(
+              "__processed_module", AnyType::get(), preprocessed_module);
+
+          // This is for the method_compile_spec passed in to to_<backend> or
+          // loaded from an exported model.
+          loweredModule.register_attribute(
+              "__method_compile_spec",
+              any_dict_ty,
+              toIValue(method_compile_spec, any_dict_ty).toGenericDict());
+
+          // This is a pointer to a backend instance that is used to access
+          // compile and execute functions via the prim::backend_compile and
+          // prim::backend_create primitives.
+          loweredModule.register_attribute(
+              "__backend", CapsuleType::get(), IValue::make_capsule(backend));
+
+          // This is the list of opaque backend handles returned by
+          // Backend::compile.
+          loweredModule.register_attribute(
+              "__handles",
+              any_dict_ty,
+              c10::impl::GenericDict(
+                  any_dict_ty->getKeyType(), any_dict_ty->getValueType()));
+
+          // Methods.
+          loweredModule.define(R"(
+            def __getstate__(self):
+                return self.__method_compile_spec, self.__processed_module
+            )");
+
+          loweredModule.define(
+              "\ndef __create_backend(self):\n\treturn ops.prim.backend_create(\"" +
+              backend_creator_name.qualifiedName() +
+              "\", self.__processed_module)\n");
+
+          loweredModule.define(R"(
+            def __setstate__(self, state):
+                self.__method_compile_spec = state[0]
+                self.__processed_module = state[1]
+                self.__create_backend()
+                self.__handles = ops.prim.backend_compile(self.__backend, self.__processed_module, self.__method_compile_spec)
+            )");
+
+          // This loop generates one method on the LoweredModule for every key
+          // in method_compile_spec.
+          for (auto& e : method_compile_spec) {
+            std::string method_name = py::cast<std::string>(e.first);
+
+            loweredModule.define(
+                "\ndef " + method_name +
+                "(self, input):\n\treturn ops.prim.backend_execute(self.__backend, self.__handles[\"" +
+                method_name + "\"], input)\n");
```

Comment:
Same here, CodeTemplate if possible

Reply:
Thanks, didn't know about that, will look into it.